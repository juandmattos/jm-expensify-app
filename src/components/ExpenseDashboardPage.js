import React from 'react'
import ReactDOM from 'react-dom'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseList />
    <ExpenseListFilters />
  </div>
)

export default ExpenseDashboardPage
