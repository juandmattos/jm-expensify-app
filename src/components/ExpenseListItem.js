import React from 'react'
import { Link } from 'react-router-dom'

export const ExpenseListItem = ({ id, description, amount, createdAt, removeExpense, history }) => (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{ `Expense: ${description}` }</h3>
      </Link>
      <p>{ `Amount: ${amount} - CreatedAt: ${createdAt}` }</p>
    </div>
)

export default ExpenseListItem
