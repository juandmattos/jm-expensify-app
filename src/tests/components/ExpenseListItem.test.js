import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseListItem } from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

test('should render ExpenseListList with expense', () => {
  const wrapper = shallow(<ExpenseListItem expense={expenses[2]}/>)

  expect(wrapper).toMatchSnapshot()
})