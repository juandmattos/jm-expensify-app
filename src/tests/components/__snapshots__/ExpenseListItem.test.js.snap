// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListList with expense 1`] = `
<div>
  <Link
    to="/edit/undefined"
  >
    <h3>
      Expense: undefined
    </h3>
  </Link>
  <p>
    Amount: undefined - CreatedAt: undefined
  </p>
</div>
`;
